gradle/scripts/lib/java-javadoc.gradle

```diff
@@ -2,15 +2,40 @@ import java.security.MessageDigest
 
 def offlineJavadoc = rootProject.hasProperty('offlineJavadoc')
 def javadocCacheDir = new File(gradle.gradleUserHomeDir, 'caches/package-lists')
-def visitedUrls = new HashSet<String>()
+def cacheDirName = 'caches/package-lists'
 
 task cleanJavadocCache(type: Delete, group: 'Build', description: 'Deletes the javadoc cache directory.') {
     delete javadocCacheDir
 }
 
+Map<String, String> javadocLinkMap = [:]
+if (JavaVersion.current() >= JavaVersion.VERSION_11) {
+    // Javadoc in Java 11+ has more strict checks related with module system,
+    // so we have to use Java 11+ API docs.
+    javadocLinkMap.put('java11', 'https://docs.oracle.com/en/java/javase/11/docs/api/')
+} else {
+    // Javadoc in pre-Java 11 generates a broken link for Java 11+ API docs,
+    // so we have to use Java 10 (or less) API docs.
+    javadocLinkMap.put('java10', 'https://docs.oracle.com/javase/10/docs/api/')
+}
+allprojects {
+    project.ext.javadocLinks.each {
+        javadocLinkMap.put(
+                "${it['groupId']}/${it['artifactId']}".toString(),
+                normalizeJavadocUrl(it['url'].toString())
+        )
+    }
+}
+
+def cacheDir = rootProject.buildDir.toPath().resolve(cacheDirName).toFile()
```

Comment:
So did we decide to move our cache to `armeria/build/caches` from ~/.gradle/caches`?

Reply:
I think so.

From what I understood, the original reason for keeping the cache in the gradle directory was so that builds in different directories can take advantage of the same cache.
I feel there is no need to do this anymore, since the gradle local/remote cache will handle this instead.

Reply:
`armeria/build/caches` is not a long-lived cache. 
Q) Do we always use Gradle remote cache to download if the input parameters are the same even for clean builds?

Reply:
> Q) Do we always use Gradle remote cache to download if the input parameters are the same even for clean builds?

My understanding is that: 
- as long as the inputs for a task didn't change, the task outputs can be fetched from the local/remote build cache. - - - `clean` doesn't have any impact on the cache as seen from experiment 3.

Let me do some testing to confirm this though 😄

Reply:
What @jrhee17 wrote is correct. This will (given that it's configured correctly) work the same way as caching of any other task.